### ISSUE TITLE
upath dependency needs updating for yarn to work

### ISSUE THREAD

**Issue by SturmB** [JeffreyWay/laravel-mix#1687]
- Laravel Mix Version: 2.1.11
- Node Version: 10.4.1
- NPM Version: 6.1.0
- OS: macOS High Sierra 10.13.5

### Description:
It seems that this version of Laravel Mix has a dependency on `upath@1.0.2` (from `chokidar`). This version of `upath` apparently does not like the latest version(s) of `Node.js` and must be updated to at least 1.0.5:

```
 ~/Sites/acs   master ●✚  yarn
yarn install v1.7.0
[1/4] 🔍  Resolving packages...
⠁ (node:90919) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
[2/4] 🚚  Fetching packages...
error upath@1.0.2: The engine "node" is incompatible with this module. Expected version ">=4 <=9".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
 ✘  ~/Sites/acs   master ●✚  yarn why upath
yarn why v1.7.0
[1/4] 🤔  Why do we have the module "upath"...?
[2/4] 🚚  Initialising dependency graph...
⠁ (node:90958) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "upath@1.0.2"
info Reasons this module exists
   - "laravel-mix#chokidar" depends on it
   - Hoisted from "laravel-mix#chokidar#upath"
   - Hoisted from "laravel-mix#webpack-dev-server#chokidar#upath"
info Disk size without dependencies: "56KB"
info Disk size with unique dependencies: "56KB"
info Disk size with transitive dependencies: "56KB"
info Number of shared dependencies: 4
✨  Done in 4.83s.
 ~/Sites/acs   master ●✚  yarn versions
yarn versions v1.7.0
{ yarn:
   '1.7.0',
  http_parser:
   '2.8.0',
  node:
   '10.4.1',
  v8:
   '6.7.288.45-node.7',
  uv:
   '1.20.3',
  zlib:
   '1.2.11',
  ares:
   '1.14.0',
  modules:
   '64',
  nghttp2:
   '1.32.0',
  napi:
   '3',
  openssl:
   '1.1.0h',
  icu:
   '61.1',
  unicode:
   '10.0',
  cldr:
   '33.0',
  tz:
   '2018c' }
✨  Done in 0.06s.
 ~/Sites/acs   master ●✚  
```

### Steps To Reproduce:

1. Create a new Laravel project per the usual instructions.
2. Install Laravel Mix.
3. Update `Node.js`, `npm`, and `yarn` to their latest versions as of this post (see above for the exact numbers).
4. Attempt to run `yarn`.

**Comment by ankurk91** [comment]
Laravel mix has a deendency of `chokidar`  `^2.0.3`

https://github.com/JeffreyWay/laravel-mix/blob/9fbd97915bfab2027dfb8f80ceac484915f5c88d/package.json#L39

Which is the latest version.
https://github.com/paulmillr/chokidar/releases

You need to re-install your node_modules from scratch to get latest version of `upath`

**Comment by SturmB** [comment]
> You need to re-install your node_modules from scratch to get latest version of upath

Done.

```
 ✘  ~/Sites/acs   master ●✚  rm -rf node_modules && npm install

> fsevents@1.2.4 install /Users/apache/Sites/acs/node_modules/fsevents
> node install

[fsevents] Success: "/Users/apache/Sites/acs/node_modules/fsevents/lib/binding/Release/node-v64-darwin-x64/fse.node" already installed
Pass --update-binary to reinstall or --build-from-source to recompile

> uws@9.14.0 install /Users/apache/Sites/acs/node_modules/uws
> node-gyp rebuild > build_log.txt 2>&1 || exit 0


> node-sass@4.9.0 install /Users/apache/Sites/acs/node_modules/node-sass
> node scripts/install.js

Cached binary found at /Users/apache/.npm/node-sass/4.9.0/darwin-x64-64_binding.node

> uglifyjs-webpack-plugin@0.4.6 postinstall /Users/apache/Sites/acs/node_modules/webpack/node_modules/uglifyjs-webpack-plugin
> node lib/post_install.js


> node-sass@4.9.0 postinstall /Users/apache/Sites/acs/node_modules/node-sass
> node scripts/build.js

Binary found at /Users/apache/Sites/acs/node_modules/node-sass/vendor/darwin-x64-64/binding.node
Testing binary
Binary is fine

> gifsicle@3.0.4 postinstall /Users/apache/Sites/acs/node_modules/gifsicle
> node lib/install.js

  ✔ gifsicle pre-build test passed successfully

> mozjpeg@5.0.0 postinstall /Users/apache/Sites/acs/node_modules/mozjpeg
> node lib/install.js

  ✔ mozjpeg pre-build test passed successfully

> optipng-bin@3.1.4 postinstall /Users/apache/Sites/acs/node_modules/optipng-bin
> node lib/install.js

  ✔ optipng pre-build test passed successfully

> pngquant-bin@4.0.0 postinstall /Users/apache/Sites/acs/node_modules/pngquant-bin
> node lib/install.js

  ✔ pngquant pre-build test passed successfully
added 1725 packages from 910 contributors and audited 19878 packages in 167.619s
found 17 vulnerabilities (5 low, 12 moderate)
  run `npm audit fix` to fix them, or `npm audit` for details
 ~/Sites/acs   master ●✚  yarn
yarn install v1.7.0
[1/4] 🔍  Resolving packages...
⠁ (node:20059) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
[2/4] 🚚  Fetching packages...
error upath@1.0.2: The engine "node" is incompatible with this module. Expected version ">=4 <=9".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
 ✘  ~/Sites/acs   master ●✚ 
```

Still the same problem.

The strange thing is that this problem does _not_ appear in another Laravel project when I run `yarn`. It's clearly something to do with my project, but I have no clue how to track it down. I'll try asking in Larachat.

**Comment by SturmB** [comment]
Fixed.

It was simply a matter of removing `node_modules`, `package-lock.json`, and `yarn.lock`, then running _only_ `yarn`, skipping the `npm install` step that I usually do.

Bottom line: abandon npm entirely. yarn ftw. 😜

**Comment by emjayess** [comment]
I just encountered this, and like the earlier report I had one Laravel project working fine (meaning `npm run watch` worked flawlessly) and another Laravel project that I'd been maintaining to closely track the first in terms of the dev setup, but where `npm run watch` kept failing with the error reported here. Eventually and reluctantly I did try blowing away /node_modules and `npm install`'d again... but with same failing results.

At closer inspection of each project's package.json, the functioning project had `browser-sync` and `browser-sync-webpack-plugin` among the `devDependencies` hash; I don't recall adding those manually, and so I don't recall who or what put them in there... but manually adding these two packages to the failing project seems to've appeased the _npm gods_, for now. 🤷‍♂️
